### DISCUSSION TITLE
Running the scoring server locally

### DISCUSSION THREAD

**Discussion by Sushant-Chavan** [hsr-project/general#8] [Q&A] [answered]
Hi,
I am a member of a team participating in the robocup-at-home-2021 challenge.

Does anyone know if the scoring server can be set up and run locally on our PC rather than on the devrt server? If yes, can someone guide us to any documentation for the same? That would help us avoid overloading the scoring server for testing small bugfixes/updates.

Thanks and regards,
Sushant Chavan

**Comment by yosuke** [comment, marked as answer]
You can log in to the build service from this page via your GitHub account and register your own worker.

https://devrt.tk/build/

This feature is still experimental, so please let me know if you experience any issues.

**Reply by Sushant-Chavan** [reply]
Thank you for the information. I tried setting up the worker and in general, it seemed to work fine. :) For someone else who is trying to do the same, once the `devrt-worker` is set up on your local PC, you need to start it by running the command `sudo devrt-worker`.

I also noticed that although the setup instruction mentions the installation of `nvidia-docker2` as optional, the simulations were not being executed on my PC and resulted in errors. However, after I installed the `nvidia-docker2` and restarted my PC, the local server seems to work fine.

@yosuke I think presently, even if we register a personal worker, the default public worker also seems to pick up these jobs, resulting in the changes being built and tested twice (as seen [here](https://github.com/Sushant-Chavan/robocup-at-home-2021-challenge-1/commit/02dbb09b3fe0422234edecdfb1202a34e78d5f1c)). Maybe this could be updated to reduce the load on the public scoring server?

**Reply by yosuke** [reply]
Thank you for the feedback (and thank you for your care on our simulation infrastructure).
In case the global simulation server is free, the task will be consumed by the server in 50% of the chance.
In this case, you don't have to care because the simulation server is in low usage.

Overlapping simulation runs are something we have to improve (thanks for the report), we will update the algorithm and let you know once it is done.

**Reply by yosuke** [reply]
It seems the overlapping simulation runs happen when the task is submitted to the empty queue and there is multiple free workers.
If the queue is not empty or the worker has a job, there will be no overlap.
Need to debug the transaction logic of the database update.

**Reply by yosuke** [reply]
The overlapping problem should be fixed now.
Can be updated by `apt-get update && apt-get upgrade`.
